Add Generation to App Data Source resource (#415)
heroku/terraform-provider-heroku: docs/data-sources/app.md

Use this data source to get information about a Heroku app.

> This data source supports both Cedar (classic buildpacks) and Fir (Cloud Native Buildpacks) generation apps.

Reviewer: ```suggestion
This data source supports both the [Cedar and Fir generations](https://devcenter.heroku.com/articles/generations) of Heroku apps.
```
